boot/cmdline.go

```diff
@@ -108,3 +110,70 @@ func MockProcCmdline(newPath string) (restore func()) {
 		procCmdline = oldProcCmdline
 	}
 }
+
+func getBootloaderManagingItsAssets(where string, opts *bootloader.Options) (bootloader.ManagedAssetsBootloader, error) {
+	bl, err := bootloader.Find(where, opts)
+	if err != nil {
+		return nil, fmt.Errorf("internal error: cannot find managed assets bootloader under %q: %v", where, err)
+	}
+	mbl, ok := bl.(bootloader.ManagedAssetsBootloader)
+	if !ok {
+		// the bootloader cannot manage its scripts
+		return nil, nil
```

Comment:
Returning nil when there's no error could be confusing, one could assume that the pointer is valid if no error is returned. Maybe we could return an error when the bootloader cannot manage scripts, and check for this error in caller?

Reply:
I will second this, you could also return `bool, obj, error` if that is more natural in the call sites.